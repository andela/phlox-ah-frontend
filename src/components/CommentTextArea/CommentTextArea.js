import React from 'react';
import PropTypes from 'prop-types';
import './CommentTextArea.scss';

const CommentTextArea = ({ handleChange, value }) => (
    <div className="comment-box">
    <textarea name="comment" value={value} onChange={handleChange} rows="10" placeholder="Comment here..."></textarea>
    </div>
);

CommentTextArea.propTypes = {
  handleChange: PropTypes.func,
  value: PropTypes.string
};

export default CommentTextArea;
