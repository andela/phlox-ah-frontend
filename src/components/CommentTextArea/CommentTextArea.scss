.comment-box {
  width: 100%;
  textarea {
    width: 100%;
    border: #E5E5E5 solid 1px;
    padding: 13px;
    font-size: 16px;
    resize: none;
    border-radius: 3px;
    height: 200px;
    &::placeholder {
      color: #C4C4C4;
      opacity: 1;
      font-size: 18px;
    }
  }
}