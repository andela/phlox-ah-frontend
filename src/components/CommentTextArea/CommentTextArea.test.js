import React from 'react';
import { mount } from 'enzyme';
import CommentTextArea from './CommentTextArea';

let component;

beforeEach(() => {
  component = mount(<CommentTextArea handleChange={() => { }} value={'This is article comment'} />);
});

afterEach(() => {
  component.unmount();
});

it('should render without throwing an error', () => {
  expect(component).toMatchSnapshot();
});

it('should have a div', () => {
  expect(component.find('div').length).toEqual(1);
});

it('should have a div with a className of comment-box', () => {
  expect(component.find('div').hasClass('comment-box'));
});

it('should have a textarea', () => {
  expect(component.find('textarea').length).toEqual(1);
});

it('should have a props of handleChange that is a function', () => {
  expect(typeof component.props().handleChange).toBe('function');
});

it('should have a props of value that is a string', () => {
  expect(typeof component.props().value).toBe('string');
});

it('should have a props of value', () => {
  expect(component.props().value).toEqual('This is article comment');
});
