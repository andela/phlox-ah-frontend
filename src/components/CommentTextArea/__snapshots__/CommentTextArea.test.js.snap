// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<CommentTextArea
  handleChange={[Function]}
  value="This is article comment"
>
  <div
    className="comment-box"
  >
    <textarea
      id="comment"
      onChange={[Function]}
      placeholder="Comment here..."
      rows="10"
      value="This is article comment"
    />
  </div>
</CommentTextArea>
`;
